client/my-sites/checkout/checkout-thank-you/redesign-v2/products/jetpack-search-product.tsx

```diff
@@ -0,0 +1,56 @@
+import { Button } from '@automattic/components';
+import { translate } from 'i18n-calypso';
+import ThankYouProduct from 'calypso/components/thank-you-v2/product';
+import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
+import { useSelector } from 'calypso/state';
+import {
+	getJetpackSearchCustomizeUrl,
+	getJetpackSearchDashboardUrl,
+} from 'calypso/state/sites/selectors';
+import { getSelectedSite } from 'calypso/state/ui/selectors';
+import type { ReceiptPurchase } from 'calypso/state/receipts/types';
+
+export type ThankYouJetpackSearchProductProps = {
+	purchase: ReceiptPurchase;
+	siteId: number | null;
+};
+
+export default function ThankYouJetpackSearchProduct( {
+	purchase,
+	siteId,
+}: ThankYouJetpackSearchProductProps ) {
+	const selectedSite = useSelector( getSelectedSite );
+	const jetpackSearchCustomizeUrl = useSelector( ( state ) =>
+		getJetpackSearchCustomizeUrl( state, siteId )
+	);
+	const jetpackSearchDashboardUrl = useSelector( ( state ) =>
+		// At this point in the flow, having purchased a product for a specific
+		// site, we can assume that `siteId` is a number and not `null`.
+		getJetpackSearchDashboardUrl( state, siteId as number )
+	);
+
+	const productButtonLabel = selectedSite?.jetpack
+		? translate( 'Go to Search Dashboard' )
+		: translate( 'Customize Search' );
+	const productButtonHref = selectedSite?.jetpack
+		? jetpackSearchDashboardUrl
+		: jetpackSearchCustomizeUrl;
+	const thankYouEventProps = selectedSite?.jetpack
+		? { product_name: 'search', value: 'Search Dashboard', site: 'jetpack' }
+		: { product_name: 'search', value: 'Customizer', site: 'wpcom' };
+
+	const recordThankYouClick = () => {
+		recordTracksEvent( 'calypso_jetpack_product_thankyou', thankYouEventProps );
+	};
```

Comment:
```suggestion

	const recordThankYouClick = () => {
		recordTracksEvent( 'calypso_jetpack_product_thankyou', {
			product_name: 'search',
			value: selectedSite?.jetpack ? 'Search Dashboard' : 'Customizer',
			site: selectedSite?.jetpack ? 'jetpack' : 'wpcom',
		} );
	};
```

What if we conditionally set the `value` and `site` event properties? That way, we avoid conditionally defining the entire object.